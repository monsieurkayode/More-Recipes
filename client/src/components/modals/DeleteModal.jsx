import React from 'react';
import PropTypes from 'proptypes';

/**
 * DeleteModal
 * @function DeleteModal
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const DeleteModal = props => (
  <div id="modal-delete" className="modal">
    <div className="modal-content">
      <h5><i className="fa fa-warning" /> {props.dialog}</h5>
      <p>This action cannot be undone!</p>
    </div>
    <div className="modal-footer">
      <button className="modal-action modal-close edit chip">
          Cancel
      </button>
      <button
        onClick={props.handleAction.bind(// eslint-disable-line
          null, props.id
        )}
        className="modal-action modal-close delete chip"
      >
        <i className="fa fa-trash" />
        { props.selected === 'favorites' ? ' Remove' : ' Delete'}
      </button>
    </div>
  </div>
);

DeleteModal.propTypes = {
  id: PropTypes.number.isRequired,
  handleAction: PropTypes.func.isRequired,
  selected: PropTypes.string.isRequired,
  dialog: PropTypes.string.isRequired
};

export default DeleteModal;
