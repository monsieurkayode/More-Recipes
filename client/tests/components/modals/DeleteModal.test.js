/* eslint-disable no-undef */
import React from 'react';

import DeleteModal from '../../../src/components/modals';

const setup = (selected, dialog) => {
  const handleAction = jest.fn();
  const props = {
    id: 1,
    handleAction,
    selected,
    dialog
  };
  const wrapper = mount(<DeleteModal {...props} />);
  return {
    wrapper,
    props
  };
};

describe('The DeleteModal Component', () => {
  it('should render correctly', (done) => {
    const { wrapper } = setup('recipes', 'Delete recipe');
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });

  it('should show Delete text on button if deleting a recipe', (done) => {
    const { wrapper } = setup('recipes', 'Delete recipe');
    expect(wrapper.find('button').last().text()).toContain('Delete');
    done();
  });

  it('should show Remove text on button if removing a favorite', (done) => {
    const { wrapper } = setup('favorites', 'Remove from favorite');
    expect(wrapper.find('button').last().text()).toContain('Remove');
    done();
  });

  it('should call handleAction once when action button is clicked', (done) => {
    const { wrapper, props } = setup('favorites', 'Remove from favorite');
    wrapper.find('button').last().simulate('click');
    expect(props.handleAction.mock.calls.length).toBe(1);
    done();
  });
});
