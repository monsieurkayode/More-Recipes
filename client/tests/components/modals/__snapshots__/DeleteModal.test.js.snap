// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DeleteModal Component should render correctly 1`] = `
<DeleteModal
  dialog="Delete recipe"
  handleAction={[MockFunction]}
  id={1}
  selected="recipes"
>
  <div
    className="modal"
    id="modal-delete"
  >
    <div
      className="modal-content"
    >
      <h5>
        <i
          className="fa fa-warning"
        />
         
        Delete recipe
      </h5>
      <p>
        This action cannot be undone!
      </p>
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="modal-action modal-close edit chip"
      >
        Cancel
      </button>
      <button
        className="modal-action modal-close delete chip"
        onClick={[Function]}
      >
        <i
          className="fa fa-trash"
        />
         Delete
      </button>
    </div>
  </div>
</DeleteModal>
`;
